src/github/github.service.ts

```diff
@@ -0,0 +1,167 @@
+      const githubUser = await this.userRepository.findOne({
+        where: { githubUserId: userId },
+      });
+      const userDetails = await this.getUserDetails(
+        githubUser.githubAccessToken,
+      );
+      return userDetails;
+    } catch (error) {
+      throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  async saveToken(userId: number, accessToken: string) {
+    const user = await this.userRepository.findOne({
+      where: { githubUserId: `${userId}` },
+    });
+    if (!user) {
+      throw new HttpException(ApiMessages.NOT_FOUND, HttpStatus.NOT_FOUND);
+    }
+    user.accessToken = accessToken;
+    await this.userRepository.save(user);
+  }
+
+  async findUserById(userId: string) {
+    return await this.userRepository.findOne({
+      where: { githubUserId: userId },
+    });
+  }
+
+  async handleInstallation(req: TRequest, installationId: string) {
+    try {
+      const userId = req.user['userId'];
+      const user = await this.userRepository.findOne({
+        where: { githubUserId: userId },
+      });
+
+      if (!user) {
+        throw new HttpException('User not found', HttpStatus.NOT_FOUND);
+      }
+      await this.validateAndUpdateInstallations(user);
+      const existingInstallation =
+        await this.userInstallationRepository.findOne({
+          where: { installationId: installationId },
+        });
+
+      if (existingInstallation) {
+        return { message: 'Installation ID up-to-date' };
+      }
+
+      const installationToken =
+        await this.generateInstallationAccessToken(installationId);
+      const repositories =
+        await this.fetchOrganizationDetails(installationToken);
+
+      if (!repositories || repositories.length === 0) {
+        throw new HttpException('No repositories found', HttpStatus.NOT_FOUND);
+      }
+
+      const organizationName = repositories[0].owner.login;
+
+      let userInstallation = await this.userInstallationRepository.findOne({
+        where: { user: { id: user.id }, organizationName: organizationName },
+      });
+
+      if (!userInstallation) {
+        userInstallation = new UserInstallation();
+        userInstallation.organizationName = organizationName;
+        userInstallation.installationId = installationId;
+        userInstallation.user = user;
+      } else if (userInstallation.installationId !== installationId) {
+        userInstallation.installationId = installationId;
+      } else {
+        return { message: 'Installation ID already up-to-date' };
+      }
+
+      await this.userInstallationRepository.save(userInstallation);
+      return { message: 'Installation ID saved/updated successfully' };
+    } catch (error) {
+      if (error.status === '404') {
+        throw new NotFoundException('App not installed');
+      }
+      throw new HttpException(
+        `Failed to handle installation: ${error.message}`,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
```

Comment:
PullProofAI

In the `handleInstallation` function, the comparison `error.status === '404'` is incorrect. It should compare against a number instead of a string.

### Improvement 3

```diff
- if (error.status === '404') {
+ if (error.status === HttpStatus.NOT_FOUND) {
```

### Issue 4



<!-- This is an auto-generated comment by PullProofAI -->